mlir/lib/Dialect/Vector/Transforms/VectorEmulateNarrowType.cpp

```diff
@@ -336,30 +442,178 @@ struct ConvertVectorStore final : OpConversionPattern<vector::StoreOp> {
     // vector.store %bitcast, %alloc[%linear_index] : memref<16xi8>,
     // vector<4xi8>
 
-    auto origElements = op.getValueToStore().getType().getNumElements();
-    if (origElements % scale != 0)
-      return failure();
+    auto origElements = valueToStore.getType().getNumElements();
+    bool isAlignedEmulation = origElements % numSrcElemsPerDest == 0;
```

Comment:
Please double check, but I think that this definition of "aligned" is only covering point 1. from https://github.com/llvm/llvm-project/blob/5a90168fa34e15b544d7749ca4d247a16d360119/mlir/lib/Dialect/Vector/Transforms/VectorEmulateNarrowType.cpp#L1072-L1079

If you agree, I'll try to propose something - we should use consistent definition throughout this file.

Reply:
Yes please! That will make subsequent changes in the file easier, and I can do the update.

Reply:
* #123526 
* #123527 
* #123528 
* #123529 

You should be able to re-use `isSubByteVecFittable ` from #123529.